components/organisms/Footer/footer.tsx

```diff
@@ -58,12 +58,31 @@ const footerContext = [
 ];
 
 const Footer = (): JSX.Element => {
+  function getSocialMediaPlatformName(url: string) {
+    switch (true) {
+      case url.includes("twitter.com"):
+        return "Twitter";
+      case url.includes("github.com"):
+        return "GitHub";
+      case url.includes("instagram.com"):
+        return "Instagram";
+      case url.includes("youtube.com"):
+        return "YouTube";
+      case url.includes("discord.com"):
+        return "Discord";
+      case url.includes("dev.to"):
+        return "Dev.to";
+      default:
+        return "Social Media";
+    }
+  }
+
```

Comment:
Refactor (blocking): No need to have a function for this. Just hard code the names in the links below, e.g.

```html
            <a
              target="_blank"
              href={url}
              key={index}
              rel="noopener noreferrer"
              aria-label={`Visit our Dev.to profile`}
            >
...
```

Reply:
> Refactor (blocking): No need to have a function for this. Just hard code the names in the links below, e.g.

Thank you for your feedback. I understand your point, and I initially considered hard-coding the social media names in the links. In this specific code block, we are mapping through the 'socials' array, which contains various social media profiles. 

Each profile has a unique URL and icon, and we want to provide accessible labels that accurately represent each profile.

```
{footerContext[2].socials?.map(({ url, icon }, index) => (
            <a
              target="_blank"
              href={url}
              key={index}
              rel="noopener noreferrer"
              aria-label={`Visit our ${getSocialMediaPlatformName(url)} profile`}
            >
              {icon}
            </a>
          ))}
```

By using the 'getSocialMediaPlatformName' function, we ensure that the accessibility labels are dynamically generated based on the specific social media URL being linked. This allows us to maintain consistency between the labels and URLs while accommodating future changes to our social media profiles.

Please let me know your thoughts.

Reply:
> > Refactor (blocking): No need to have a function for this. Just hard code the names in the links below, e.g.
> 
> Thank you for your feedback. I understand your point, and I initially considered hard-coding the social media names in the links. In this specific code block, we are mapping through the 'socials' array, which contains various social media profiles.
> 
> Each profile has a unique URL and icon, and we want to provide accessible labels that accurately represent each profile.
> 
> ```
> {footerContext[2].socials?.map(({ url, icon }, index) => (
>             <a
>               target="_blank"
>               href={url}
>               key={index}
>               rel="noopener noreferrer"
>               aria-label={`Visit our ${getSocialMediaPlatformName(url)} profile`}
>             >
>               {icon}
>             </a>
>           ))}
> ```
> 
> By using the 'getSocialMediaPlatformName' function, we ensure that the accessibility labels are dynamically generated based on the specific social media URL being linked. This allows us to maintain consistency between the labels and URLs while accommodating future changes to our social media profiles, this approach keeps our code DRY (Don't Repeat Yourself) and maintainable.
> 
> Please let me know your thoughts.

Reply:
I'm just seeing how that page was originally implemented, i.e. `footerContext`. Instead of `getSocialMediaPlatformName(url)`, add the label text to the `footerContext` directly.

```
footerContext = [
...
{
    socials: [
      {
        url: "https://twitter.com/saucedopen",
        icon: <FaXTwitter className="text-2xl hover:text-light-slate-10  text-light-slate-9" />,
				label: "OpenSauced on Twitter"
      },
			....
    ],
  },
```

and then in the JSX:

```jsx
          {footerContext[2].socials?.map(({ url, icon, label }, index) => (
            <a target="_blank" aria=label={label} href={url} key={index} rel="noopener noreferrer">
              {icon}
            </a>
          ))}
```

Reply:
+1 on this @nickytonline, it would be more cleaner this way 👍

Reply:
> I'm just seeing how that page was originally implemented, i.e. `footerContext`. Instead of `getSocialMediaPlatformName(url)`, add the label text to the `footerContext` directly.
> 
> ```
> footerContext = [
> ...
> {
>     socials: [
>       {
>         url: "https://twitter.com/saucedopen",
>         icon: <FaXTwitter className="text-2xl hover:text-light-slate-10  text-light-slate-9" />,
> 				label: "OpenSauced on Twitter"
>       },
> 			....
>     ],
>   },
> ```
> 
> and then in the JSX:
> 
> ```js
>           {footerContext[2].socials?.map(({ url, icon, label }, index) => (
>             <a target="_blank" aria=label={label} href={url} key={index} rel="noopener noreferrer">
>               {icon}
>             </a>
>           ))}
> ```

Oh Great! 
Thank you, @nickytonline. This is perfect 😃.

Reply:
Done